react/ProductListContext.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { createContext, useContext, useReducer, FC } from 'react'
+
+interface State {
+  products: any[]
```

Comment:
Isn't a way to type this?

Reply:
@klzns and I were discussing about this and we think that right now this may not be necessary because:
1 - On the shelf we know how a product is, but on the `ProductList` app it may have a different structure
2 - Type checking won't matter much on this occasion since we only want to store them